### PR TITLE
add possiblity to pass baseOptions to the API Configuration, which will be passed to every fetch call

### PR DESCRIPTION
**Pull request by dsumer**
…ll be passed to every fetch call

### PR checklist

- [X] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [X] Ran the shell script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh` and `./bin/security/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates). Windows batch files can be found in `.\bin\windows\`.
- [X] Filed the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `3.2.x`, `4.0.x`. Default: `master`.
- [X] Copied the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

@TiFu (2017/07) @taxpon (2017/07) @sebastianhaas (2017/07) @kenisteward (2017/07) @Vrolijkx (2017/09) @macjohnny (2018/01)

### Description of the PR

add possiblity to pass baseOptions to the API Configuration, which will be passed to every fetch call

useful to pass e.g. { credentials: 'include' } as baseOptions and don't have to pass it to every single API call

### CODE REVIEW

**Comment by dsumer** [comment]
hm the shippable job timed out, do you have any tips / insights on this?

**Comment by wing328** [comment]
Hi @TiFu @taxpon @sebastianhaas  @kenisteward  @Vrolijkx @macjohnny, any question/feedback on this PR?

File: samples/client/petstore-security-test/typescript-fetch/api.ts [hunk 1]
```diff
@@ -17,7 +17,7 @@ import * as url from "url";
 import * as portableFetch from "portable-fetch";
 import { Configuration } from "./configuration";
 
-const BASE_PATH = "https://petstore.swagger.io *_/ ' \" =end -- \\r\\n \\n \\r/v2 *_/ ' \" =end -- \\r\\n \\n \\r".replace(/\/+$/, "");
+const BASE_PATH = "http://petstore.swagger.io *_/ ' \" =end -- \\r\\n \\n \\r/v2 *_/ ' \" =end -- \\r\\n \\n \\r".replace(/\/+$/, "");
```
[the remark this answers is not in the document]
Comment (dsumer) [reply samples/client/petstore-security-test/typescript-fetch/api.ts]: yes seems like they were not regenerated before, honestly i don't know much about it, it changed after executing the sample binaries

**Comment by wing328** [comment]
@dsumer thanks for the contribution, which has been merged into master.

@macjohnny @TiFu thanks for reviewing the change.

**Comment by wing328** [comment]
@dsumer thanks for the enhancement, which is included in the v3.2.2 stable release: https://twitter.com/oas_generator/status/1032252335131512832

**Pull request merged**